transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/AntibodyTransformations.scala

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.monster.encode.transformation
+
+import java.time.OffsetDateTime
+import org.broadinstitute.monster.encode.jadeschema.table.Antibody
+import org.slf4j.LoggerFactory
+import upack.Msg
+
+object AntibodyTransformations {
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  val EncodeIdPattern = "/[^/]+/([^/]+)/".r
+
+  def transformAntibody(antibodyInput: Msg): Antibody = {
+    import org.broadinstitute.monster.common.msg.MsgOps
+
+    val id = antibodyInput.read[String]("accession")
+    val targets = antibodyInput
+      .read[Array[String]]("targets")
+
+    // Use regular expressions to remove everything but the actual target name, which should be the same across
+    // all targets in the list. Remove "synthetic_tag" type targets.
+    val mappedTargets = targets.collect {
```

Comment:
This `collect` can become a `map`:
```
targets.map(CommonTransformations.readId)
```
We lose the filter-if-it-doesn't-match behavior, but that's probably a good thing; if we get malformed target strings, we probably want to go boom.

Reply:
Would the `readId` method specifically be applicable here? It looks like it takes in a message and searches for the `@id` field, whereas we want to just read in a string (like `"/targets/.../"`).

I think the `transformId` method would be more fitting, right?